jobs/bump-lockfile.Jenkinsfile

```diff
@@ -35,86 +36,140 @@ try { lock(resource: "bump-${params.STREAM}") { timeout(time: 120, unit: 'MINUTE
     """)
 
     def branch = params.STREAM
-    def timestampOnly = false
+    def forceTimestamp = false
+    def haveChanges = false
     shwrap("cosa init --branch ${branch} https://github.com/${repo}")
     shwrap("cosa buildprep ${BUILDS_BASE_HTTP_URL}/${branch}/builds")
 
-    def prevPkgChecksum = shwrapCapture("jq -c .packages src/config/manifest-lock.x86_64.json | sha256sum")
-    def prevPkgTimestamp = shwrapCapture("jq -c .metadata.generated src/config/manifest-lock.x86_64.json | xargs -I{} date --date={} +%s") as Integer
-
+    def lockFile, PkgChecksum, PkgTimestamp
+    def archinfo = [x86_64: [:], aarch64: [:]]
+    for (arch in archinfo.keySet()) {
+        lockFile = "src/config/manifest-lock.${arch}.json"
+        if (utils.path_exists("${lockFile}")) {
+            PkgChecksum = shwrapCapture("jq -c .packages ${lockFile} | sha256sum")
+            PkgTimestamp = shwrapCapture("jq -c .metadata.generated ${lockFile} | xargs -I{} date --date={} +%s") as Integer
+            archinfo[arch]['prevPkgChecksum'] = PkgChecksum
+            archinfo[arch]['prevPkgTimestamp'] = PkgTimestamp
+        } else {
+            archinfo[arch]['prevPkgChecksum'] = ""
+            archinfo[arch]['prevPkgTimestamp'] = 1
+        }
+    }
     // do a first fetch where we only fetch metadata; no point in
     // importing RPMs if nothing actually changed
     stage("Fetch Metadata") {
-        shwrap("cosa fetch --update-lockfile --dry-run")
+        parallel x86_64: {
+            shwrap("cosa fetch --update-lockfile --dry-run")
+        }, aarch64: {
+            def gangparams = gangplank.getGangPlankFCOSAARCH64Params()
+            gangparams['extraFlags'] += ' --returnFiles=src/config/manifest-lock.aarch64.json'
+            gangparams['cmd'] = "cosa fetch --update-lockfile --dry-run"
+            gangplank.gangPlankFCOSAARCH64BuilderWrapper(gangparams)
+            shwrap("cp builds/cache/src/config/manifest-lock.aarch64.json src/config/manifest-lock.aarch64.json")
+        }
     }
 
-    def newPkgChecksum = shwrapCapture("jq -c .packages src/config/manifest-lock.x86_64.json | sha256sum")
-    def newPkgTimestamp = shwrapCapture("jq -c .metadata.generated src/config/manifest-lock.x86_64.json | xargs -I{} date --date={} +%s") as Integer
-    if (newPkgChecksum == prevPkgChecksum) {
-        println("No changes")
-        if ((newPkgTimestamp - prevPkgTimestamp) > (2*24*60*60)) {
+    for (arch in archinfo.keySet()) {
+        lockFile = "src/config/manifest-lock.${arch}.json"
+        PkgChecksum = shwrapCapture("jq -c .packages ${lockFile} | sha256sum")
```

Comment:
I'd recommend a function to read the lock file and return the `PkgChecksum` and `PkgTimestamp`

Reply:
added a function for `getLockfileInfo()` for this.